### DISCUSSION TITLE
Compilation of observations and issues

### DISCUSSION THREAD

**Discussion by ShreyanJabade** [fvutils/pyvsc#80] [General]
Hello,
With the latest release, the nested classes no longer give a syntactical error. Thank you for your consideration in this aspect. 
I further tested all the other features of PyVSC together with the nested classes (as will be required in the general use-case). Have documented the observations and issues I encountered in a doc linked herewith. Have also provided a link to a Colab Python notebook cell where the relevant aspect is tested with an example for ready reference. Few general aspects (i.e. not specifically related to nesting) are highlighted in yellow (e.g. unwanted solutions due to overflow). Will update the doc if anything else is encountered.
I request you to consider these points.
Link to doc: https://docs.google.com/document/d/1udAE9E6fO1qoq4DSeAWTBERG_b2-ULMVPj5wasGaM1s/edit?usp=sharing
Thank you

**Comment by mballance** [comment]
Hi @ShreyanJabade,
    Thank you for doing this further investigation and collecting the results.
    There are two things that you can do that would be very helpful to me. First, if you could file a GitHub Issue for each of the cases you've uncovered and include the testcase in the issue report, that would help in tracking which issues are open vs which have been resolved. Second, and perhaps more important, if you could prioritize the issues (either low/medium/high or in sequence 1..N), that would help me to resolve the ones most important to you first. 

Thanks and Best Regards,
Matthew

**Reply by ShreyanJabade** [reply]
Hello,
Have created a new separate Github issue for each of the aspects. Have provided the detailed description, code snippet and the output/error for each of the issues. These testcases are essentially extensions of the nested testcase in our previous discussion. 
Thank you

**Comment by mballance** [comment]
Many thanks, @ShreyanJabade. I'll begin working my way through the issues, starting with the highest-priority.

Best Regards,
Matthew

**Comment by mballance** [comment]
Hello @ShreyanJabade,
    I wanted to provide you an update on these issues, and an ETA on completing them.
- #81 and #82 -- These are corrected in the 0.4.3 release
- #92, #91, #88, #86, #85, #84 -- These are corrected in the 0.4.4 release

    There are three remaining issues
- #83 -- High bias in cases of nested constraints
- #90 -- References to dynamic constraints declared in arrays
- #89 -- Bias with dynamic constraints

    All three of these will require more investigation, with the 'bias' bugs taking the most time. Given the data I currently have, I estimate being able to close the remaining issues by the end of this week (7/3/2021). I'll investigate #83 first. Please advise if this isn't in line with your priorities.

Best Regards,
Matthew

**Reply by ShreyanJabade** [reply]
Thank you for your consideration of all the issues. Yes, #83 is in line with the priorities.
